client/indexer-service/src/handler.rs

```diff
@@ -239,11 +240,12 @@ impl<Runtime: StorageEnableRuntime> IndexerService<Runtime> {
                 value_prop_id: _,
             } => {
                 let old_msp = if let Some(id) = old_msp_id {
-                    Some(Msp::get_by_onchain_msp_id(conn, id.to_string()).await?)
+                    Some(Msp::get_by_onchain_msp_id(conn, OnchainMspId::from(*id)).await?)
                 } else {
                     None
                 };
-                let new_msp = Msp::get_by_onchain_msp_id(conn, new_msp_id.to_string()).await?;
+                let new_msp =
+                    Msp::get_by_onchain_msp_id(conn, OnchainMspId::from(*new_msp_id)).await?;
```

Comment:
Maybe for better ergonomics I could have the parameter type be `impl Into<OnchainMspId>`, that way you any callers of these db model functions can just pass the provider ids directly